src/expr/src/linear.rs

```diff
@@ -1359,33 +1363,29 @@ pub mod plan {
                             .chain(None.into_iter());
                     }
                     Ok(Datum::Numeric(d)) => {
-                        let v = i128::try_from(d.0).unwrap();
-                        if lower_bound_i128 < v {
-                            lower_bound_i128 = v;
+                        // Negative values will always be less than `time`, so
+                        // will never be chosen.
+                        if d.0.is_negative() {
+                            continue;
                         }
-                    }
-                    Ok(Datum::Null) => {
-                        null_eval = true;
-                    }
-                    x => {
-                        panic!("Non-decimal value in temporal predicate: {:?}", x);
-                    }
-                }
-            }
 
-            // If there are any upper bounds, determine the minimum upper bound.
-            for u in self.upper_bounds.iter() {
-                match u.eval(datums, &arena) {
-                    Err(e) => {
-                        return Some(Err((e, time, diff)))
-                            .into_iter()
-                            .chain(None.into_iter());
-                    }
-                    Ok(Datum::Numeric(d)) => {
-                        // Replace `upper_bound` if it is none
-                        let v = i128::try_from(d.0).unwrap();
-                        if upper_bound_i128.is_none() || upper_bound_i128 > Some(v) {
-                            upper_bound_i128 = Some(v);
+                        match u64::try_from(d.0) {
+                            Ok(v) => {
+                                let lb = lower_bound
+                                    .expect("lower bound is only none on exit from loop");
+                                if lb < v {
+                                    lower_bound = Some(v);
+                                }
+                            }
+                            // If conversion fails, value is greater than
+                            // u64::MAX, which isn't comparable to time, so set
+                            // lower bound to None. This would have the greatest
+                            // value we have seen, so do not continue looking
+                            // for a larger lower bound.
+                            Err(_) => {
+                                lower_bound = None;
+                                break;
+                            }
```

Comment:
I'm not sure I follow this. Setting `lower_bound` to `None` probably causes the rest of the code to behave as if there is no lower bound, rather than a very large lower bound, right?

Reply:
No, it seems not. The meaning of `None` for the two bounds (lower, upper) both seems to be "a large value" rather than "not bounded". I'll probably follow with a PR with some explanatory text ahead of the code.

Reply:
Yes, the behavior is "large value." This was what `None` indicated in the previous iteration, though the check was not placed at this location. Making the lower bound `None` here let me avoid an interstitial `lower_bound_numeric` value..